src/integrations/manageLinesInFile.ts

```diff
@@ -0,0 +1,94 @@
+import fs from 'fs';
+import os from 'os';
+import isEqual from 'lodash/isEqual.js';
+
+export const START_LINE = '### MANAGED BY RANCHER DESKTOP START (DO NOT EDIT)';
+export const END_LINE = '### MANAGED BY RANCHER DESKTOP END (DO NOT EDIT)';
+const DEFAULT_FILE_MODE = 0o644;
+
+// Inserts/removes fenced lines into/from a file. Idempotent.
+// @param path The path to the file to work on.
+// @param desiredManagedLines The lines to insert into the file.
+// @param desiredPresent Whether the lines should be present.
+export async function manageLinesInFile(path: string, desiredManagedLines: string[], desiredPresent: boolean): Promise<void> {
+  // read file, creating it if it doesn't exist
+  let currentContent: string;
+
+  try {
+    currentContent = await fs.promises.readFile(path, 'utf8');
+  } catch (error: any) {
+    if (error.code === 'ENOENT' && desiredPresent) {
+      const lines = buildFileLines([], desiredManagedLines, []);
+      const content = lines.join(os.EOL);
```

Comment:
Yes I think that would be the case. Is that alright?

Reply:
As long as we decide it is, it should be fine?
I do know that some _really_ old compilers and things would miss the last line (because they were looking for the line feed), but that's like, 80s or earlier, I think.